sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -107,20 +107,24 @@ namespace Azure { namespace Core { namespace Test {
     {
       std::string updated(src);
       // Remove special marker for LIVEONLY
-      auto const noPrefix
+      auto noPrefix
           = std::regex_replace(updated, std::regex(TestContextManager::LiveOnlyToken), "");
+      // Remove special marker for PLAYBACKONLY
+      noPrefix = std::regex_replace(updated, std::regex(TestContextManager::PlaybackOnlyToken), "");
```

Comment:
I don't follow this logic. You are saving the value of noPrefix but only care if it matches in the later case?

Reply:
this code is kind of stupid, filed bug https://github.com/Azure/azure-sdk-for-cpp/issues/5466 , some refactroring is needed, 
since we needed to stop the hsm thing i did not do it here. 
the idea is that when the name is sanitized/read if a token was removed , since we care only about recorded /live, we need to check the test context to determine if the test is allowed to run in the scenario.
i would argue that should be all in the context  not here, it was there before and refactoring could cause unforseen issues in other packages thus my desire to limit the blast radius of this PR.